app/src/org/commcare/views/widgets/CalendarFragment.java

```diff
@@ -0,0 +1,342 @@
+                }
+            }
+        });
+
+        today.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                calendar.setTimeInMillis(todaysDateInMillis);
+                refresh();
+            }
+        });
+    }
+
+    //Redraws the calendar display
+    public void refresh(){
+        ArrayList<Date> dateList = new ArrayList<>();
+        Calendar populator = (Calendar) calendar.clone();
+
+        int totalDays = getNumDaysInMonth(populator);
+        populateListOfDates(dateList, populator, totalDays);
+
+        yearSpinner.setSelection(calendar.get(Calendar.YEAR)-GregorianDateWidget.MINYEAR);
+        monthSpinner.setSelection(calendar.get(Calendar.MONTH));
+        myGrid.setAdapter(new CalendarAdapter(getContext(), dateList));
+    }
+
+    //Populates an arraylist with dates
+    private void populateListOfDates(ArrayList<Date> dateList, Calendar populator, int totalDays) {
+        while(dateList.size() < totalDays){
+            dateList.add(populator.getTime());
+            populator.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        int remainingDays = ((DAYSINWEEK + 1)-(populator.get(Calendar.DAY_OF_WEEK)))%DAYSINWEEK;
+
+        for(int i = 0; i < remainingDays; i ++){
+            dateList.add(populator.getTime());
+            populator.add(Calendar.DAY_OF_MONTH, 1);
+        }
+    }
+
+    //Calculates the total number of days from most recent Sunday to the first Saturday following the end of the current month to fill the calendar view
+    private int getNumDaysInMonth(Calendar cal){
+        int totalDays = cal.getActualMaximum(Calendar.DAY_OF_MONTH);
+        cal.set(Calendar.DAY_OF_MONTH, 1);
+
+        //Day of week for the first of the month
+        int monthStartWeekDay = cal.get(Calendar.DAY_OF_WEEK) - 1;
+        int daysSincePreviousSunday = -monthStartWeekDay;
+
+        totalDays += monthStartWeekDay;
+
+        //Backtracking calendar to the most recent Sunday
+        cal.add(Calendar.DAY_OF_MONTH, daysSincePreviousSunday);
+
+        return totalDays;
+    }
+
+    private class CalendarAdapter extends ArrayAdapter<Date>{
+
+        private LayoutInflater mInflater;
+
+        public CalendarAdapter(Context context, ArrayList<Date> dates){
+            super(context, R.layout.calendar_date, dates);
+            mInflater = LayoutInflater.from(context);
+        }
+
+        @Override
+        public View getView(int position, View view, ViewGroup parent){
+            if(view == null){
+                view = mInflater.inflate(R.layout.calendar_date, null);
+            }
+
+            TextView text = (TextView) view;
+
+            Date date = getItem(position);
+            text.setText(String.valueOf(date.getDate()));
+
+            Date current = calendar.getTime();
+
+            if(date.equals(current)){
+                text.setTextColor(getResources().getColor(R.color.white));

```

Comment:
`getColor(int)` is deprecated, should use `getColor(int, Theme)` and just pass `null` in for the `Theme` (as described in the docs of the `getColor` method)